defaults.go

```diff
@@ -18,7 +18,7 @@ package platforms
 
 // DefaultString returns the default string specifier for the platform.
 func DefaultString() string {
-	return Format(DefaultSpec())
+	return FormatAll(DefaultSpec())
```

Comment:
This is changing defaults here.

/cc @tonistiigi @tianon

Reply:
That _seems_ expected to me, but highlights a possible issue.

A quick survey of the containerd code calling this function suggests that by-and-large it's used to hold a string that is given back to `Parse` later; so in the same process it's fine, and in fact due to the removal of auto-insertion of Windows host OS version in Parse, this one _needs_ to be `FormatAll` or we're changing the observed behaviour of `Parse(DefaultString())` on Windows.

The Windows v2 Runtime manager's supported platforms list might need to be hard-coded though. It already hard-codes `linux/amd64` (that should probably be `linux/$(GOARCH)`...?).

Some cases however are in the client library, and perhaps the string is being processed in containerd on the other side of the (stable API!) remote connection. _If_ that's the case then depending on the direction of mismatch, we're either generating strings old containerd can't parse, or generating strings which new containerd will parse into a platform object unexpectedly absent OS version.

I haven't checked to confirm that the stable API actually contains these platform-as-strings, the ones I was able to trace through GitHub's WebUI _seemed_ to still be `Parse`d in the client library code, so maybe this problematic case doesn't exist here.

Direct consumers of this library will be affected by the defaults change, but AFAIK this library is not 1.0-versioned?

Reply:
@dmcgowan should we cut "1.x" tag from this repo or change DefaultString() to use Format() and have a new function that uses FormatAll() ?